framework-cloud/framework-openshift-operator/src/main/java/org/kie/cloud/openshift/operator/deployment/KieServerOperatorDeployment.java

```diff
@@ -42,8 +43,13 @@ public KieServerOperatorDeployment(Project project, NonNamespaceOperation<KieApp
     public void scale(int instances) {
         if (isReady()) {
             KieApp kieApp = kieAppClient.withName(OpenShiftConstants.getKieApplicationName()).get();
-            Server server = getAssociatedServerObject(kieApp);
+
+            Spec appliedSpec = kieApp.getStatus().getApplied();
+            Server server = getAssociatedServerObject(appliedSpec);
             server.setReplicas(instances);
+
+            // Update current spec
+            kieApp.setSpec(appliedSpec);
```

Comment:
I did this way to make clear that we're getting an immutable copy of the spec and using it to update the current spec of the component.

Reply:
here as we are only changing the replicas, I think it's ok to change only Spec and we not need to get all generated stuff that's it in applied. The default properties should be applied again.

Reply:
The problem is that we need to update only the servers with a concrete name (from getServiceName()). 
In most of the scenarios, the name is populated automatically by the operator and driven by the environment (and it's in the status.applied). So, if the name is not supplied in the scenario builder, we would be fine to set the replicas as you suggested. But the problem is that in some other scenarios, we set the name in scenario builder (to have kie-server1 and kie-server2 for example) and we would only need to update one, not all.
Therefore, to cope with the both scenarios, I decided to use the specification from the status.applied, set the replicas and update the kieapp.spec. 

Another approach could be to detect if we have named servers, if so, search by name, if not, we scale all the servers. This works but can be trickier.